### PR TITLE
Updated project reference to SQL DacFx and Added Unit Tests

### PR DESCRIPTION
**Pull request by MarkEwer**
I updated the primary Nuget reference that was importing the SQL ScriptDom to use the newer SQL DacFx package instead. This package is compatible with netStandard 2.0 so this switch should make it possible to convert the overall project to use netStandard and, therefore, DotNetCore.

Also, I added unit tests for NULLIF, ISNULL, and string comparison operations.

### CODE REVIEW

File: PocketSql.Lib/Engine.cs [hunk 1]
```diff
@@ -9,7 +9,9 @@ public class Engine
     {
         private readonly SqlVersion sqlVersion;
 
-        public Engine(int version) : this(IntToSqlVersion(version)) { }
+        public Engine(int version) : this(IntToSqlVersion(version))
```
Comment (rkoeninger) [inline PocketSql.Lib/Engine.cs]: Unnecessary whitespace change

File: PocketSql.Lib/EngineCommand.cs [hunk 2]
```diff
@@ -35,8 +35,12 @@ public IDbConnection Connection
 
         // Command control doesn't do anything
         public void Dispose() { }
-        public void Prepare() { }
-        public void Cancel() { }
+        public void Prepare()
```
Comment (rkoeninger) [inline PocketSql.Lib/EngineCommand.cs]: Unnecessary whitespace change

File: PocketSql.Lib/EngineResult.cs [hunk 3]
```diff
@@ -4,7 +4,9 @@ namespace PocketSql
 {
     public class EngineResult
     {
-        public EngineResult() { }
+        public EngineResult()
```
Comment (rkoeninger) [inline PocketSql.Lib/EngineResult.cs]: Unnecessary whitespace change

File: PocketSql.Lib/EngineTransaction.cs [hunk 4]
```diff
@@ -14,8 +14,12 @@ public EngineTransaction(EngineConnection connection, IsolationLevel il)
 
         // Transactions don't do anything
         public void Dispose() { }
-        public void Commit() { }
-        public void Rollback() { }
+        public void Commit()
```
Comment (rkoeninger) [inline PocketSql.Lib/EngineTransaction.cs:17]: Unnecessary whitespace change

File: PocketSql.Tests/EvaluationTests.cs [hunk 5]
```diff
@@ -6,7 +6,7 @@
 
 namespace PocketSql.Tests
 {
-    [TestFixture]
+  [TestFixture]
```
Comment (rkoeninger) [inline PocketSql.Tests/EvaluationTests.cs]: Unnecessary whitespace change

File: PocketSql.Tests/SqlFunctionsForNullValueTests.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
+﻿using System.Data;
```
Comment (rkoeninger) [inline PocketSql.Tests/SqlFunctionsForNullValueTests.cs:1]: Use 4 space indent instead of 2

File: PocketSql.Tests/WhereConditionTests.cs [hunk 7]
```diff
@@ -0,0 +1,60 @@
+﻿using System.Data;
```
Comment (rkoeninger) [inline PocketSql.Tests/WhereConditionTests.cs:1]: Use 4 space indent instead of 2

File: PocketSql.Tests/app.config [hunk 8]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (rkoeninger) [inline PocketSql.Tests/app.config]: Is this app.config needed for was it automatically created by visual studio?

**Review by rkoeninger — CHANGES_REQUESTED** [review]
Thanks for the PR!

Have you tried DacFx with Core? Is it advertised that way somewhere? I see a `net46` folder in the nuget package only, and no `netstandard2_0` or anything like that.

I left some comments, mostly about style. I would suggest using a GUI Git client (I still use GitHub for Desktop) so you can easily see all of the changes that are staged and commit only the files/lines that you need. It's also easier to rollback individual files that way.

**Comment by MarkEwer** [comment]
Sorry, I use a code analysis / cleaning plugin as part of my day job to enforce company conventions (like braces on separate lines and 2-space tabs).  I'll try to pay more attention if I need to send in an additional PR.

File: PocketSql.Lib/Engine.cs (hunk 1 above)
Comment (MarkEwer) [reply PocketSql.Lib/Engine.cs]: fixed.

File: PocketSql.Lib/EngineCommand.cs (hunk 2 above)
Comment (MarkEwer) [reply PocketSql.Lib/EngineCommand.cs]: fixed.

File: PocketSql.Lib/EngineResult.cs (hunk 3 above)
Comment (MarkEwer) [reply PocketSql.Lib/EngineResult.cs]: fixed.

File: PocketSql.Lib/EngineTransaction.cs (hunk 4 above)
Comment (MarkEwer) [reply PocketSql.Lib/EngineTransaction.cs:17]: fixed.

File: PocketSql.Tests/SqlFunctionsForNullValueTests.cs (hunk 6 above)
Comment (MarkEwer) [reply PocketSql.Tests/SqlFunctionsForNullValueTests.cs:1]: fixed.

File: PocketSql.Tests/EvaluationTests.cs (hunk 5 above)
Comment (MarkEwer) [reply PocketSql.Tests/EvaluationTests.cs]: fixed.

File: PocketSql.Tests/WhereConditionTests.cs (hunk 7 above)
Comment (MarkEwer) [reply PocketSql.Tests/WhereConditionTests.cs:1]: fixed.

File: PocketSql.Tests/app.config (hunk 8 above)
Comment (MarkEwer) [reply PocketSql.Tests/app.config]: fixed.

**Comment by MarkEwer** [comment]
> Have you tried DacFx with Core? Is it advertised that way somewhere? I see a `net46` folder in the nuget package only, and no `netstandard2_0` or anything like that.

Yes, and yes.  If you go to https://www.nuget.org/packages/Microsoft.SqlServer.DACFx/150.4573.2 and look under the dependencies you should see that it has support for NetStandard 2.0.

You don't get that package downloaded by default unless one of the projects in the solution is targeting NetStandard.  If you only have full framework libraries, then Nuget will only get the full framework version.

**Comment by rkoeninger** [comment]
> look under the dependencies you should see that it has support for NetStandard 2.0.

I did not see that earlier. Awesome! Now I can stop working on re-compiling the ScriptDom dll.

Thanks for cleaning up the formatting stuff.

Merging.

**Review by rkoeninger — APPROVED** [review]
